### PR TITLE
#687 - Combobox multiselect

### LINKED ISSUE #687: [Combobox] Implement a multi-select component accordingly to the specifications
Part of #620 

Implement a multiselect-select component, using the `useCombobox` hook, accordingly to the [specifications](https://www.figma.com/file/TE9FUDtlgVQ4FWlAPtTagxQU/Bold-Design-System?node-id=3750%3A44).

### PR DESCRIPTION
**Pull request by Andrevmatias**
Closes #687

### CODE REVIEW

File: src/components/Combobox/Combobox.stories.tsx [hunk 1]
```diff
@@ -153,3 +150,30 @@ export const CustomComponents = () => (
     }}
   />
 )
+
+export const MultiSelect = () => (
+  <Combobox<Fruit>
+    itemIsEqual={(a, b) => a.label === b.label}
+    value={fruits.filter(
+      (e) =>
+        optionsKnob('values', Object.fromEntries(fruits.map((e) => [e.label, e.label])), [], {
```
Comment (gustavomoser) [inline src/components/Combobox/Combobox.stories.tsx:159]: I went to the branch's deployment to see how it works and.. it doesn't seem to be working :sweat_smile:

File: src/components/Combobox/ComboboxSingleselect.tsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  filter?(items: T[], filter: string): T[]
+  onChange?: (newValue: T) => void
+  onFilterChange?: (newValue: string) => void
+  components?: Partial<ComboboxComponents<T>>
+  multiple: false
+
+  inputId?: string
+  labelId?: string
+  menuId?: string
+  getItemId?(idx: number): string
```
Comment (gustavomoser) [inline src/components/Combobox/ComboboxSingleselect.tsx:35]:
```suggestion
  getItemId?(index: number): string
```

**Review by gustavomoser — APPROVED** [review]
wonderful!! :clap: I left only two comments :smile:

**Review by romulofer — APPROVED** [review]

File: src/components/Combobox/Combobox.stories.tsx (hunk 1 above)
Comment (Andrevmatias) [reply src/components/Combobox/Combobox.stories.tsx]: You're right, it wasn't working, but now it does :grin:

File: src/components/Combobox/ComboboxMultiselect.tsx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+      zIndex: theme.zIndex.dropdown,
+      border: `1px solid ${theme.pallete.divider}`,
+      borderRadius: theme.radius.popper,
+      backgroundColor: theme.pallete.surface.main,
+      boxShadow: theme.shadows.outer['40'],
+      maxHeight: '20rem',
+    } as CSSProperties,
+
+    list: {
+      zIndex: 'auto',
```
Comment (calistro) [inline src/components/Combobox/ComboboxMultiselect.tsx:309]: Isn't 'auto' the default value for z-index already?
```suggestion
```

**Review by calistro — APPROVED** [review]
:heart:

File: src/components/Combobox/ComboboxMenuComponents.tsx [hunk 4]
```diff
@@ -36,31 +40,82 @@ export interface ComboboxComponents<T> {
...
   )
+})
+
+export const ComboboxMultiselectMenuItem = React.forwardRef(
+  (props: ComboboxMultiselectItemProps<any>, ref: Ref<HTMLLIElement>) => {
+    const { children, item, style, selected, highlighted, itemToString, index, ...rest } = props
+    const { classes, css } = useStyles(createStyles)
+
+    return (
+      <li ref={ref} className={css(classes.item, highlighted && classes.selected, style)} {...rest}>
```
[the remark this answers is not in the document]
Comment (Andrevmatias) [reply src/components/Combobox/ComboboxMenuComponents.tsx:88]: There is no indication that the selected (checked checkbox) items should have the same style as the highlighted (current "cursor" position or mouse hover) one, neither in the prototypes nor in the previous Select component. I think it's valid only for the `ComboboxSingleselect`, but I've already done it in the `ComboboxInline` PR.

**Pull request merged**